### PR TITLE
SAK-42821 reduce polling to /portal/timeout and /direct/session

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- library/src/morpheus-master/js/src/sakai.morpheus.session.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakai/sakai/pullRequest?prid=4564736&bid=15346800)

**Comment by fostersdesign** [comment]
Does this change affect the default of `timeoutSeconds@org.sakaiproject.presence.api.PresenceService`? If so, that property's default value should be updated in default.sakai.properties.

**Comment by ottenhoff** [comment]
Hi @fostersdesign presence isn't affected here

File: library/src/morpheus-master/js/src/sakai.morpheus.session.js [hunk 1]
```diff
@@ -61,9 +56,23 @@ var poll_session_data = function(){
...
+              .done(function(htmlSegment) {
+                timeoutDialogFragment = htmlSegment;
+                show_timeout_alert(min);
+              })
+              .fail(function() {
+                timeoutDialogEnabled = false;
+              });
+          }
+          else {
+                show_timeout_alert(min);
```
[the remark this answers is not in the document]
Comment (ottenhoff) [reply library/src/morpheus-master/js/src/sakai.morpheus.session.js]: agree thanks

File: library/src/morpheus-master/js/src/sakai.morpheus.session.js [hunk 2]
```diff
@@ -61,9 +56,23 @@ var poll_session_data = function(){
...
+            $PBJQ.ajax({ url: "/portal/timeout?auto=true", cache: true, dataType: "text"})
+              .done(function(htmlSegment) {
+                timeoutDialogFragment = htmlSegment;
+                show_timeout_alert(min);
+              })
+              .fail(function() {
+                timeoutDialogEnabled = false;
+              });
+          }
+          else {
```
[the remark this answers is not in the document]
Comment (ottenhoff) [reply library/src/morpheus-master/js/src/sakai.morpheus.session.js]: agree

File: portal/portal-charon/charon/src/webapp/scripts/portalscripts.js [hunk 3]
```diff
@@ -73,130 +73,6 @@ function removeDHTMLMask() {
...
-			sessionTimeOut = setTimeout("poll_session_data()", 1000 * 10);
-		}
-	},
-	error: function(XMLHttpRequest, status, error){
-		// We used to to 404 handling here but now we should always get good session data.
-	}
-	});
-}
-
-function keep_session_alive(){
```
[the remark this answers is not in the document]
Comment (ottenhoff) [reply portal/portal-charon/charon/src/webapp/scripts/portalscripts.js]: ok will delete timeout.vm

**Comment by ottenhoff** [comment]
> Just get rid of timeout.vm

Hmmmm seems to be used by TimeoutDialogHandler.java

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- library/src/morpheus-master/js/src/sakai.morpheus.session.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakaiproject/sakai/pullRequest?prid=4564736&bid=15346800)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- library/src/morpheus-master/js/src/sakai.morpheus.session.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sakaiproject/sakai/pullRequest?prid=4564736&bid=15346800)

**Pull request merged**